source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -46,27 +46,43 @@ Common::Redis::Client::PoolRequest* makeSingleServerRequest(
   }
   return handler;
 }
-AdminRespHandlerType getresponseHandlerType(const std::string command_name) {
+AdminRespHandlerType getresponseHandlerType(const std::string command_name, const std::string command_arg = "") {
+  std::string keyspacepattern = "__keyspace@0__";
+  std::string keyeventpattern = "__keyevent@0__";
   AdminRespHandlerType responseHandlerType = AdminRespHandlerType::response_handler_none;
   if (Common::Redis::SupportedCommands::allShardCommands().contains(command_name)) {
     if (command_name == "pubsub" || command_name == "keys" || command_name == "slowlog" || command_name == "client" || command_name == "info") {
       responseHandlerType = AdminRespHandlerType::aggregate_all_responses;  
-    } else if (command_name == "script" || command_name == "flushall" || command_name == "config" || command_name == "select"){
+    } else if (command_name == "script" || command_name == "flushall" || command_name == "config" || command_name == "select") {
       responseHandlerType = AdminRespHandlerType::allresponses_mustbe_same;
+    } else if (command_name == "subscribe" || command_name == "psubscribe") {
+      if (command_arg.substr(0, keyspacepattern.length()) == keyspacepattern || command_arg.substr(0, keyeventpattern.length()) == keyeventpattern) {
+        responseHandlerType = AdminRespHandlerType::allresponses_mustbe_same;
+      } else {
+        responseHandlerType = AdminRespHandlerType::singleshardresponse;
+      }
+    } else if (command_name == "unsubscribe" || command_name == "punsubscribe") {
+      responseHandlerType = AdminRespHandlerType::aggregate_all_responses;
+    } else {
+      responseHandlerType = AdminRespHandlerType::singleshardresponse;
     }
   }else if (command_name == "publish" || command_name == "cluster"){
     responseHandlerType = AdminRespHandlerType::singleshardresponse;
   }
   return responseHandlerType;
 }
-int32_t getShardIndex(const std::string command, int32_t requestsCount,int32_t redisShardsCount) {
+int32_t getShardIndex(const std::string command, int32_t requestsCount,int32_t redisShardsCount, const std::string command_arg = "") {
   int32_t shard_index = -1;
+  std::string keyspacepattern = "__keyspace@0__";
+  std::string keyeventpattern = "__keyevent@0__";
   //This need optimisation , generate random seed only once per thread
   srand(time(nullptr));
   
   if (Common::Redis::SupportedCommands::blockingCommands().contains(command) && requestsCount == 1) {
     return shard_index;
-  }else if (!Common::Redis::SupportedCommands::allShardCommands().contains(command) && requestsCount == 1 ){
+  } else if (!Common::Redis::SupportedCommands::allShardCommands().contains(command) && requestsCount == 1 && 
+   !( (command == "subscribe" || command == "psubscribe") && 
```

Comment:
this complex if else needs to be cleaned up
why cant it simply be another if condifiton 
if subscribe commands and keyspace pattern then return -1